### DISCUSSION TITLE
Plan to implement mtv rooms creation

### DISCUSSION THREAD

**Discussion by Devessier** [AdonisEnProvence/MusicRoom#112] [Ideas]
As I listed in the description of [Mtv room creation](https://github.com/AdonisEnProvence/MusicRoom/projects/6) project, I think we could cut the development in several parts, allowing us to implement business logic without worrying too much about technical details, such as geolocation.

The main parts I discern are the following ones:

- Room creation wizard form
- Persist room settings in the workflow
- Only invited users can vote
- Only users at a given position and at a given moment can vote
- Geolocate the user and report its position to the workflow

I intentionally put geolocation at the end of the list. We should be able to simulate that information in other developments.
Each of these parts will be described in an issue.

Let me know what you think about this plan.

**Comment by prastoin** [comment]
Nice,
I think this plan is really good first step in this room creation journey

We also might want to list here what could be every possible settings added to the room creation.

Room Creation:
The creator can define few settings during the room's creation:
Note: ( Playlists later could be used to init a room )

- The name of the room, ?? which is necessarily unique ?? ( I feel like we could add an index at the end of a room name during the adonis process easily )
- Choose room's status: private or public ( it said default public in the subject, we could use a toggle input ? )
- If the room is public: Can toggle allow only invited users to vote to the playlist
- Can set custom geoloc and dates/times constraints to authorize the vote to specific located users. The creator needs to set both a date and a location.
- Also choose a mode, direct or broadcast.
- Minimum score needed for a track to be played

![MTVSUBJECT](https://user-images.githubusercontent.com/45004772/131626526-f3d084e3-4772-44e9-bd7e-2a839506ee33.png)
